onnxruntime/core/graph/contrib_ops/collective_defs.cc

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/graph/contrib_ops/contrib_defs.h"
+#include "core/graph/constants.h"
+
+namespace onnxruntime {
+namespace contrib {
+
+using ONNX_NAMESPACE::AttributeProto;
+using ONNX_NAMESPACE::InferenceContext;
+using ONNX_NAMESPACE::OpSchema;
+using ONNX_NAMESPACE::OPTIONAL_VALUE;
+using ONNX_NAMESPACE::TypeProto;
+
+void RegisterCollectiveOps() {
+  ONNX_CONTRIB_OPERATOR_SCHEMA(AllReduce)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .Input(0, "input", "tensors to be reduced", "T", OpSchema::Variadic)
+      .Output(0, "output", "reduced tensors", "T", OpSchema::Variadic)
+      .TypeConstraint(
+          "T",
+          {"tensor(float16)", "tensor(float)", "tensor(double)"},
+          "Constrain to float, float16 and double tensors.")
+      .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+        propagateShapeAndTypeFromFirstInput(ctx);
+      });
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(AllGather)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .Attr("world_size",
+            "total world size that need to be gathered.",
```

Comment:
Is "world-size" the commonly used word in these APIs? Otherwise, I would suggest a name like "num_processors" or "num_tensors" ,which I find a bit easier to understand. In any case, I would suggest updating the description to "Number of tensors to be gathered" or "Number of processors participating in the all-gather operation."

Reply:
Agree with Rama here. world_size is a word in the MPI vocab which is an implementation detail of this kernel. Can we rename this to something more generic and intuitive?